src/errors.h

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2018-2020 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#pragma once
+
+#include <stddef.h>
+#include "rax.h"
+
+// Report an error in filter placement with the first unresolved entity.
+void Error_InvalidFilterPlacement(rax *entitiesRax);
```

Comment:
Seems more appropriate to pass the actual filter tree, and report a string representation of it.
`void Error_InvalidFilterPlacement(const FT_FilterNode *filter);`

Reply:
I agree that is a more sensible signature, but the rax passed to this function consists solely of filter tree references that couldn't be resolved, which seems a lot more actionable to me than a full string representation.